nightingale_interface/main.py

```diff
@@ -0,0 +1,147 @@
+import asyncio
+import roslibpy
+import json
+
+from kivy.config import Config
+
+Config.set("graphics", "width", "1280")
+Config.set("graphics", "height", "800")
+
+from kivymd.app import MDApp
+from kivy.uix.screenmanager import ScreenManager
+import kivy
+from kivy.properties import NumericProperty
+
+import MovoConfig
+from screen_wrapper import ScreenWrapper
+from screens.screen_config import ScreenConfig as cfg
+
+
+class MainApp(MDApp, ScreenWrapper):
+    _other_task = None
+    coms_enabled = True
+
+    # ros things
+    client = None
+    ros_action_topic = None
+    interface_screen_topic = None
+
+    # task queue things
+    queue_task = []
+    queue_delay = []
+    queue_args = []
+
+    # counters for items
+    water_count = NumericProperty(0)
+    ice_count = NumericProperty(0)
+    blanket_count = NumericProperty(0)
+
+    def main(self):
+        self._other_task = asyncio.ensure_future(self.backend())
+
+        async def run_wrapper():
+            await self.async_run(async_lib="asyncio")
+            print("App done")
+            self._other_task.cancel()
+
+        return asyncio.gather(run_wrapper(), self._other_task)
+
+    # add a task to do after a delay
+    def queue(self, task, delay=0, args=None):

```

Comment:
Now that we've discussed the system as sending a status to the UI and the UI sending the user input data back, is this "queue" functionality required? i feel like the mission planner will always send the next status when it is available and will not end up queuing multiple tasks to do at a time.

Reply:
This is more for blocking tasks to keep the touch screen unblocked rather than to actually queue tasks. This allows the execution of blocking tasks from a seperate thread

Reply:
I think we should remove this now that the Mission planner is doing the actual computation and the tablet is just an input interface. What do you think?

Reply:
I disagree. Lets talk in person about this, it has made me think of more things that need to be implemented specifically related to the estop

Reply:
ok I'll leave it in for V0